services/f-droid/f-droid.service.js

```diff
@@ -55,64 +64,30 @@ module.exports = class FDroid extends BaseYamlService {
     }
   }
 
-  async fetchYaml(url, options) {
-    const yaml = await this._requestYaml({
+  async fetch({ appId }) {
+    const url = `https://gitlab.com/kitsunyan/fdroidapi/raw/master/applications/${appId}`
+    return this._requestJson({
       schema,
-      url: `${url}.yml`,
-      ...options,
-    })
-    return { version: yaml.CurrentVersion }
-  }
-
-  async fetchText(url, options) {
-    const { buffer } = await this._request({
-      url: `${url}.txt`,
-      ...options,
-    })
-    const metadata = buffer.toString()
-    // we assume the layout as provided here:
-    // https://gitlab.com/fdroid/fdroiddata/raw/master/metadata/axp.tool.apkextractor.txt
-    const positionOfCurrentVersionAtEndOfTheFile = metadata.lastIndexOf(
-      'Current Version:'
-    ) // credits: https://stackoverflow.com/a/11134049
-    const lastVersion = metadata.substring(
-      positionOfCurrentVersionAtEndOfTheFile
-    )
-
-    const match = lastVersion.match(/^Current Version:\s*(.*?)\s*$/m)
-    if (!match) {
-      throw new InvalidResponse({
-        prettyMessage: 'invalid response',
-        underlyingError: new Error('could not find version on website'),
-      })
-    }
-    return { version: match[1] }
-  }
-
-  async handle({ appId }, { metadata_format: metadataFormat }) {
-    const url = `https://gitlab.com/fdroid/fdroiddata/raw/master/metadata/${appId}`
-    const fetchOpts = {
-      options: {},
+      url,
       errorMessages: {
+        403: 'app not found',
         404: 'app not found',
       },
-    }
-    const fetch = metadataFormat === 'yml' ? this.fetchYaml : this.fetchText
-    let result
+    })
+  }
 
-    try {
-      // currently, we only use the txt format to the initial fetch because
-      // there are more apps with that format but yml is now the standard format
-      // on f-droid, so if txt is not found we look for yml as the fallback
-      result = await fetch.call(this, url, fetchOpts)
-    } catch (error) {
-      if (metadataFormat) {
-        // if the format was specified it doesn't make the fallback request
-        throw error
-      }
-      result = await this.fetchYaml(url, fetchOpts)
+  async handle({ appId }, { include_prereleases: prereleases }) {
+    const json = await this.fetch({ appId })
+    const svc = prereleases === undefined && json.suggestedVersionCode
+    const packages = (json.packages || []).filter(
+      ({ versionCode }) => !svc || versionCode <= svc
+    )
+    if (packages.length === 0) {
+      throw new NotFound({ prettyMessage: 'no packages found' })
     }
-
-    return this.constructor.render(result)
+    const version = packages.reduce((a, b) =>
+      a.versionCode > b.versionCode ? a : b
+    ).versionName
```

Comment:
FWIW a common pattern with the service classes is to extract this sort of logic into a separate `transform` function that's responsible for interrogating the raw response to extract and transform the desired data points. Besides the separation of concerns, an additional benefit is that it's a lot easier to test a `transform` function with standard unit tests vs. trying to setup all the mocks with service tests. 

Don't feel like you need to change this though, mostly sharing for awareness